### PR TITLE
New cards

### PR DESCRIPTION
**Pull request by ataliercio**
creating gridpacks for the UL production: thq, tth, thw MadGraph5 cards with ctcvcp weights

### CODE REVIEW

**Comment by agrohsje** [comment]
Dear @ataliercio. Please make sure all *~ files are deleted. I started doing so. In the proc card please remove old lines that you don't need as indicated in my examples. Suggest to use consistently 4FS and 5FS in your cards. What about useful names for all the ME reweightings?

**Comment by ataliercio** [comment]
Dear @agrohsje , 
thanks a lot for your comments, I'm adding them. But I have a couple of questions:
I'm trying to produce the correrct gridpacks for UL for the samples thq_4f_ckm_LO_ctcvcp, thw_5f_ckm_LO_ctcvcp and tth_4f_ckm_LO_ctcvcp
- thq_4f_ckm_LO_ctcvcp: I'm refering to the FullRunII gridpack (https://cms-pdmv.cern.ch/mcm/requests?prepid=HIG-RunIIFall18wmLHEGS-00623&page=0&shown=127) if you check the 'old' cards (https://github.com/cms-sw/genproductions/blob/8d97961cda6fffccfa09cc245b715ce6f2b43f6f/bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Higgs/thq_4f_ckm_LO_ctcvcp_MH/thq_4f_ckm_LO_ctcvcp_MH_run_card.dat#L73) here the asrwgtflavor is set to 5 even if in the name it's written 4f... it's just a problem of naming? Refering to your request ```Suggest to use consistently 4FS and 5FS in your cards``` could you tell me what I should do then?
Another thing requarding this card is that hgg is also producing the gridpack and they are using:
```False = pdfwgt
4 = asrwgtflavor```
do I have to uniformize? 
- thw_5f_ckm_LO_ctcvcp: the old gridpack is https://cms-pdmv.cern.ch/mcm/requests?prepid=HIG-RunIIFall18wmLHEGS-00624&page=0&shown=127 (cards: https://github.com/cms-sw/genproductions/blob/8d97961cda6fffccfa09cc245b715ce6f2b43f6f/bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Higgs/thq_4f_ckm_LO_ctcvcp_MH/thq_4f_ckm_LO_ctcvcp_MH_run_card.dat) this one is exactly the same the same card that I've modified for UL (no name conflicts), also hgg is producing this processs and the only difference between my card and their card is that they are setting:
```False = pgfwgt``` instead I'm following the RunII card and I'm using ```True = pdfwgt``` do I have to uniformize on this?
- tth_4f_ckm_LO_ctcvcp: there is the name conflict of thq_4f_ckm_LO_ctcvcp... the RunII card is https://github.com/cms-sw/genproductions/blob/master/bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Higgs/tth_4f_ckm_LO_ctcvcp/tth_4f_ckm_LO_ctcvcp_run_card.dat#L72 and if you check here the asrwgtflavor is set to 5... what should I do? Do I have to rename the cards or change the asrwgtflavor to 4? I think 5f is the correct configuration.

Thanks a lot.

**Comment by sihyunjeon** [comment]
@agrohsje Correct me if I am wrong.
@ataliercio From what I know the parameters in block [1] will only trigger when you need to do jet matchings. For the cards you pointed out (again, as far as I know) won't really matter whether asrwgtflavor is set to 5 or 4 because jet matching isn't done ==> won't do anything to your gridpacks. What's important is what's in the bottom [2].
So I am guessing Alexander is talking about your thq thw tth having un-unified flavour schemes? Some 4f, some 5f.

[1] https://github.com/cms-sw/genproductions/blob/8d97961cda6fffccfa09cc245b715ce6f2b43f6f/bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Higgs/thq_4f_ckm_LO_ctcvcp_MH/thq_4f_ckm_LO_ctcvcp_MH_run_card.dat#L67-L75
[2] https://github.com/cms-sw/genproductions/blob/8d97961cda6fffccfa09cc245b715ce6f2b43f6f/bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/Higgs/thq_4f_ckm_LO_ctcvcp_MH/thq_4f_ckm_LO_ctcvcp_MH_run_card.dat#L255

**Comment by agrohsje** [comment]
Let me clarify. @sihyunjeon is right. I usually propose to use a consistent setup to avoid that someone else picks up the cards, adds a jet and then doesn't fix the flavor scheme. But indeed (best would be to confirm again from the code) my understanding is that the alphas reweighting flavor scheme is only relevant when additional jets are produced.

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_customizecards.dat [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#put card customizations here (change top and higgs mass for example)
+set param_card mass 6 172.5
+set param_card yukawa 6 -172.5
```
Comment (sihyunjeon) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_customizecards.dat:3]: Is it set to "-172.5" on purpose? Would give you different cross sections when setting it to +172.5 right?
And is higgs mass set to what?

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_proc_card.dat [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+set complex_mass_scheme False
+import model sm
+define p = g u c d s u~ c~ d~ s~
+define j = g u c d s u~ c~ d~ s~
+define l+ = e+ mu+ ta+
+define l- = e- mu- ta-
+define vl = ve vm vt
+define vl~ = ve~ vm~ vt~
+define ll = l+ l-
+define vll = vl vl~
```
Comment (sihyunjeon) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_proc_card.dat:13]: please remove the unused lines

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_run_card.dat [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+#*********************************************************************
+# Renormalization and factorization scales                           *
+#*********************************************************************
+  False	=  fixed_ren_scale   ! if .true. use fixed ren scale
+  False	=  fixed_fac_scale   ! if .true. use fixed fac scale
+  91.188	=  scale             ! fixed ren scale
+  91.188	=  dsqrt_q2fact1     ! fixed fact scale for pdf1
+  91.188	=  dsqrt_q2fact2     ! fixed fact scale for pdf2
+  -1	=  dynamical_scale_choice  ! Select one of the preselect dynamical choice
+  0.33 =  scalefact         ! scale factor for event-by-event scales
```
Comment (sihyunjeon) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_run_card.dat:49]: I am not very knowledgeable with this setting.
Is 0.33 set with some reason?

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_customizecards.dat (hunk 1 above)
Comment (ataliercio) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_customizecards.dat:4]: We are following the same configuration used in the full RunII card. The Higgs mass is set to 125 GeV, we made it explicit in the new version of the cards.

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_run_card.dat (hunk 3 above)
Comment (ataliercio) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_run_card.dat:49]: We are following the same configuration used in the full RunII card, the purpose of the analyzer is to have exactly the same configuration for the UL cards.

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_customizecards.dat (hunk 1 above)
Comment (sihyunjeon) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_customizecards.dat:4]: I did a quick test and I think this gives different cross sections.
This should be checked I think

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_run_card.dat (hunk 3 above)
Comment (sihyunjeon) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_run_card.dat:49]: @agrohsje Are you aware of this?

Comment (sihyunjeon) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_run_card.dat:49]: But do we not need to understand what this does?

Comment (sihyunjeon) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_run_card.dat:49]: resolved

**Comment by agrohsje** [comment]
@ataliercio . can you please follow-up on the remaining comments. thanks!

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_customizecards.dat (hunk 1 above)
Comment (vischia) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_customizecards.dat:4]: Dear @sihyunjeon , can you please clarify what gives a different cross section with respect to what? The minus value for the yukawa coupling is desired and naturally yields a different cross section.

Unless you mean that setting explicitly the Higgs mass changes the cross section, to which I guess the answer is that it depends on what is the model default for the Higgs mass. In CMS production, we conventionally set it to 125 afaik, so I think @ataliercio did well in setting it explicitly to 125.

Comment (sihyunjeon) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_customizecards.dat:4]: @vischia @ataliercio 
I looked into the UFO model file and it looks like the default yukawa coupling for top quark is set to +173 (positive value). Now here you set it to -172.5 which is a negative value so my question was is this really intended as from the model card its yukawa are by default set to positive values ==> I just wanted to make sure whether this sign flipping was on purpose or not. But it seems like it does, right?

**Comment by vischia** [comment]
Hi @*sihyunjeon,*

Yes, it is on purpose! Thank you very much!

Cheers,
Pietro

**Comment by sihyunjeon** [comment]
@agrohsje this is good to go in

**Comment by agrohsje** [comment]
hi angela will you follow-up on the remaining comments? let me know. i think ul branch files will not be further used so i am also ok with merging.

**Comment by ataliercio** [comment]
hi @agrohsje if you are refering to the comment 'remove the unused lines' I uploaded couple of days ago the new version of the cards, as far as I can see there should be no more comments to be followed up..

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thw_5f_ckm_LO_ctcvcp/thw_5f_ckm_LO_ctcvcp_proc_card.dat [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+set complex_mass_scheme False
+import model sm
+define p = g u c d s u~ c~ d~ s~
+define j = g u c d s u~ c~ d~ s~
+define l+ = e+ mu+ ta+
+define l- = e- mu- ta-
+define vl = ve vm vt
+define vl~ = ve~ vm~ vt~
+define ll = l+ l-
+define vll = vl vl~
```
Comment (sihyunjeon) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thw_5f_ckm_LO_ctcvcp/thw_5f_ckm_LO_ctcvcp_proc_card.dat:13]: @agrohsje must be referring to these lines

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_proc_card.dat [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+set ignore_six_quark_processes False
+set loop_optimized_output True
+set complex_mass_scheme False
+import model sm
+define p = g u c d s u~ c~ d~ s~
+define j = g u c d s u~ c~ d~ s~
+define l+ = e+ mu+ ta+
+define l- = e- mu- ta-
+define vl = ve vm vt
+define vl~ = ve~ vm~ vt~
```
Comment (sihyunjeon) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/UL/thq_4f_ckm_LO_ctcvcp/thq_4f_ckm_LO_ctcvcp_proc_card.dat:11]: these as well, which are from the "sm" model

**Pull request merged**